x/params/types/params.go

```diff
@@ -41,3 +44,30 @@ func validateFeesParams(i interface{}) error {
 	}
 	return nil
 }
+
+func NewCosmosGasParams(multiplierNumerator uint64, multiplierDenominator uint64) CosmosGasParams {
+	return CosmosGasParams{
+		CosmosGasMultiplierNumerator:   multiplierNumerator,
+		CosmosGasMultiplierDenominator: multiplierDenominator,
+	}
+}
+
+func (cg *CosmosGasParams) Validate() error {
+	if cg.CosmosGasMultiplierDenominator == 0 {
+		return errors.New("Cosmos Gas Multiplier Denominator can not be 0")
```

Comment:
shouldnt we also assert that numerator can't be 0 to safeguard against making all txs free?

Reply:
I intentionally left that out. Denominator not 0 is required to avoid panics. Numerator can validly be 0 and it requires a gov proposal to change. I think we can leave this as is.

Reply:
Update - added in the validation check here to avoid numerator is 0 after discussion